router/pkg/mcpserver/server.go

```diff
@@ -0,0 +1,676 @@
+		return fmt.Errorf("failed to load operations: %w", err)
+	}
+
+	sseServer, httpServer, err := s.ServeSSE()
+	if err != nil {
+		return fmt.Errorf("failed to create SSE server: %w", err)
+	}
+
+	// Store server references for Stop method
+	s.sseServer = sseServer
+	s.httpServer = httpServer
+
+	// Register tools
+	s.registerTools()
+
+	return nil
+}
+
+// Stop gracefully shuts down the MCP server
+func (s *GraphQLSchemaServer) Stop(ctx context.Context) error {
+	if s.httpServer == nil {
+		return nil
+	}
+
+	s.logger.Debug("shutting down MCP server")
+
+	// Create a shutdown context with timeout
+	shutdownCtx, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
+
+	// Shutdown the HTTP server
+	if err := s.httpServer.Shutdown(shutdownCtx); err != nil {
+		return fmt.Errorf("failed to gracefully shutdown MCP server: %w", err)
+	}
+
+	return nil
+}
+
+// registerTools registers all tools for the MCP server
+func (s *GraphQLSchemaServer) registerTools() {
+	// TODO: Support tool annotations when available in MCP Go SDK
+
+	// Add a tool to list all available operations (names and descriptions only)
+	s.server.AddTool(
+		mcp.NewTool(
+			"list_operations",
+			mcp.WithDescription("Lists all available GraphQL operations."),
+		),
+		s.handleListGraphQLOperations(),
+	)
+
+	s.server.AddTool(
+		mcp.NewTool(
+			"get_operation_info",
+			mcp.WithDescription("Retrieve comprehensive metadata and execution details for a specific GraphQL operation by its name. Use this to collect all required information needed to execute the operation via execute_<operation_name>."),
+			mcp.WithString("operationName",
+				mcp.Required(),
+				mcp.Description("The exact name of the GraphQL operation to retrieve information for."),
+			),
+		),
+		s.handleGraphQLOperationInfo(),
+	)
+
+	// Only register the schema tool if exposeSchema is enabled
+	if s.exposeSchema {
+		s.server.AddTool(
+			mcp.NewTool(
+				"get_schema",
+				mcp.WithDescription("Use this function to obtain the full introspection-based schema of the GraphQL API. This is useful for understanding the structure, available types, queries, mutations, and overall capabilities of the API."),
+			),
+			s.handleGetGraphQLSchema(),
+		)
+	}
+
+	// Only register the execute_graphql tool if enableArbitraryOperations is enabled
+	if s.enableArbitraryOperations {
+		// Add a tool to execute arbitrary GraphQL queries
+		executeGraphQLSchema := []byte(`{
+			"type": "object",
+			"description": "The query and variables to execute.",
+			"properties": {
+				"query": {
+					"type": "string",
+					"description": "The GraphQL query or mutation string to execute."
+				},
+				"variables": {
+					"type": "object",
+					"additionalProperties": true,
+					"description": "The variables to pass to the GraphQL query as a JSON object."
+				}
+			},
+			"additionalProperties": false,
+			"required": ["query"]
+		}`)
+
+		s.server.AddTool(
+			mcp.NewToolWithRawSchema(
+				"execute",
```

Comment:
is the name explicit enough? Have you considered "execute_raw_graphql_operation" or similar? I'm not sure what is good naming for tools yet.